src/types.mts

```diff
@@ -0,0 +1,91 @@
+// Local Imports
+import type { configurationDefaults } from './RollbarClientSubmitter.mjs';
+
+// Local Types
+type TConfigurationOptionCustom = Omit<IConfigurationOptions, 'accessToken'>;
+
+type TConfigurationOptionsWithDefaults = IConfigurationOptions & typeof configurationDefaults;
+
+type TConfigurationSerialized<T extends object> = {
+  // eslint-disable-next-line @typescript-eslint/ban-types -- any function can be expected here
+  [key in keyof T]: Exclude<T[key], undefined | false> extends RegExp | Function ? string : T[key];
```

Comment:
This will be ever so slightly worst than what I had provided, because it will transform e.g. `RegExp | false` to `string`, while ideally it should be `string | false` (or perhaps an optional string)

Reply:
This took awhile to figure out why your code was causing errors, and I figured it out.

First off, addressing your comment, there is no field expecting `RegExp | false`, I know that must have been more confusing with the `IConfigurationOptions` and `IConfigurationInternal`; now only the prior exists. For that use case, either the expected data types are used, or a `string` is expected in the `custom.configuration` object for `RegExp` and `Function` objects so they can be serialized with `toString()`.

The reason the very clever type code you wrote was causing errors on a few fields (namely the `RegExp` and `Function` fields) is because (1) those fields were optional and (2) I have the `tsconfig` option `exactOptionalPropertyTypes` enabled which causes optional fields to have `undefined` added implicitly as part of a union type. That's why I had to explicitly `Exclude<>` `undefined`; `Required<>` did not do the trick here unexpectedly.

Given all this, what are your thoughts?

Reply:
On the code I proposed (maybe not the very first one, but the improved version I had put in a comment), I don't think `undefined` should be any problem (I tried it out in the example below too and things seem to work how I'd expect them to).

Checking also your current solution for `undefined`, it is quite interesting the results 👀 
`Function | undefined` is transformed to `string`, which isn't ideal.
On the other hand, optionality seems to work, so that `test?: Function` is transformed into `test?: string`, I am guessing `[key in keyof T]` keeps the optionality, hence why it seems to work for what is here.

It's slightly less complete than the other implementation as it won't deal properly with a field that can be a RegExp/Function or something else (e.g, `Function | number`), cf [this example](https://www.typescriptlang.org/play?#code/C4TwDgpgBAYgrgOwMbAJYHsEEEDOBlYAJ1QQHMAeAFQD4oBeKSqCAD2AgQBMdZEUMEUAPxQcREqSgAuRgG4AUKEhQAShFIBRFmFwFiZKrQZNW7LjzWbtw0eLLS58p-JLtCAMwCGSaDE8BrCAB5ACMAKwgUKABveSh4qHYxIRl4ZDRMBQTEiDEAJlS+DIQshKTgAGYZSy0wKAAfKEROCHcSCE5S+PKAFhkxfVIunLEAVhSoBDgAWxCIQgbbQcW0-kxFmu0FAF8FRXBoSjx51E8AG1QALw7QiJQqZjYObih0cMjgIxi4qABtQJAUBIUAB6HcjAAutV1LVdHYKKtinDBlR-hAQBDqNQdnslIcAMKYNqkOCETzFY7Ec5XDoPUzPHhvO6fejfeIAenZzBwFwQwAAtJxUDhPCEzhB+QhTPzedAAAJ4nBIYhgAW5XnAdkhTwIfmKqD8-lQHWA9xFARQJA6qBzR6QFAdKAAC3mEB+aMBwNB4MoUKgWiQZzgLVRAIhABomlxWu1OIsvGccBBaPTzKoYdZGoiLSIBhIHJQPRCcU5OVAAOqoYBOxJO4WvKWR85nWvQfWeQjQJDoQidlBnQFEHU4dw96YdSM4dC18kjYCjKCkCDAHisbzAAetqDEUhO4BQABu5zguX2ykppwu104fkCtw+rKOJ2p1-v91vwXeKGx8jLlertb1pgECRuUC4AO6YAA5PuS77tW0A7nuh7HrkNqRJ4cBJlA4HQB20CYJuwAdnBaD2AhohwCESb7gABs0MZSnGjQJkmtFntAF4vh0H5vsA-5OgACioACS0xgC2xiEggxKkuSAhcVetK8V+nyyEAA), but you don't have anything like that at the moment, so not critical.

Reply:
Asking question via Slack